tests/shared/src/test/scala/cats/tests/Predicate.scala

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2015 Typelevel
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package cats.tests
+
+import cats.Decidable
+import cats.kernel.Eq
+import cats.laws.discipline.eq._
+import cats.laws.discipline.ExhaustiveCheck
+import org.scalacheck.{Arbitrary, Cogen}
+
+case class Predicate[A](run: A => Boolean)
+object Predicate {
+  implicit val decidablePredicate: Decidable[Predicate] =
```

Comment:
Can we implement this by delegating to the instance in Cats itself?
https://github.com/typelevel/cats/blob/7730debb3209aa4a28db92f01dd7e8ff943d33c8/core/src/main/scala/cats/instances/function.scala#L83

Or are we testing that instance elsewhere?

Reply:
Good call, I'll definite it via delegation!

Reply:
Really that instance in `instances.function` should have a `Ring` context bound if not `Rig` but I don't think `core` depends on `algebra` does it?

Reply:
Hmm, I'm afraid not, `core` only depends on `kernel`. Why does it need to be `Ring`/`Rig`, how should it actually look like? If we're not doing it right maybe we shouldn't do it at all 😕

Reply:
I mean it's a fine instance and it's useful, but here's a way more generic one we could have, if I remember right:
```scala
implicit def catsStdDecidableForRigValuedFunctions[C: Rig] = 
  new Decidable[* => C] {
    def unit: * => C = Function.const(Rig[C].one)
    def zero: Nothing => A = Function.const(Rig[C].zero)
    def sum(fa: A => C, fb: B => C): Either[A, B]] => C = 
       _.fold(fa, fb)
     def product(fa: A => C, fb: B => C): (A, B) => C =
       t => Rig[C].combine(fa(t._1), fb(t._2))
  }
```
Which ought to generalize this one based on the `Rig` for `Boolean`, but it's not huge, it would provide a few more instances "for free".
So this instance is totally legitimate but we it's kind of the shadow of this other more fundamental instance.

Reply:
Oh I see, that is quite nice. Maybe one day ring-like stuff can find its way into Cats proper.